recipes/iridescence/all/conanfile.py

```diff
@@ -0,0 +1,152 @@
+            "apple-clang": "11",
+            "Visual Studio": "16",
+            "msvc": "192",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("assimp/5.4.2")
+        self.requires("boost/1.84.0", transitive_headers=True)
+        self.requires("eigen/3.4.0", transitive_headers=True)
+        self.requires("glm/cci.20230113")
+        self.requires("libjpeg/9e")
+        self.requires("libpng/[>=1.6 <2]")
+        self.requires("portable-file-dialogs/0.1.0", transitive_headers=True)
+        # Upstream uses -docking version of imgui, but it would cause version conflicts on CCI
+        self.requires("imgui/1.90.5", transitive_headers=True, transitive_libs=True)
+        self.requires("imguizmo/cci.20231114")
+        self.requires("implot/0.16")
+        self.requires("glfw/3.4", transitive_headers=True, transitive_libs=True)
+        self.requires("opengl/system")
+        # https://github.com/koide3/iridescence/blob/7034275ee6516eb4d155f645cd8327173edfeb9d/thirdparty/gl3w/GL/glcorearb.h#L82
+        self.requires("khrplatform/cci.20200529", transitive_headers=True)
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        if is_msvc(self):
+            # Build fails with several compilation errors
+            raise ConanInvalidConfiguration("MSVC is not supported")
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("Shared builds are not supported on Windows")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.16 <4]")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["IMGUI_BACKENDS"] = os.path.join(self.dependencies["imgui"].package_folder, "res", "bindings").replace("\\", "/")
+        tc.variables["BUILD_EXAMPLES"] = "OFF"
+        tc.variables["BUILD_PYTHON_BINDINGS"] = "OFF"
+        tc.variables["BUILD_WITH_MARCH_NATIVE"] = "OFF"
+        tc.variables["BUILD_EXT_TESTS"] = "OFF"
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
+        if "docking" not in str(self.dependencies["imgui"].ref.version):
```

Comment:
In case you ever need to do this kind of comparision again, the `docking` prefix is actually treated as a prerelease in Conan :)

```suggestion
        if self.dependencies["imgui"].ref.version.pre != "docking":
```